driver-core/src/test/unit/com/mongodb/logging/TestLoggingInterceptor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.logging;
+
+import com.mongodb.internal.logging.LogMessage;
+import com.mongodb.internal.logging.LoggingInterceptor;
+import com.mongodb.internal.logging.StructuredLogger;
+import com.mongodb.lang.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class TestLoggingInterceptor implements LoggingInterceptor, AutoCloseable {
```

Comment:
Did this change at all or was it just moved?

Reply:
This class was just moved to driver-core in order to be visible in DefaultConnectionPoolSpecification as well as in unified tests

Reply:
This class was moved to the driver-core module to ensure accessibility in both DefaultConnectionPoolSpecification and unified tests. However,  since modifications were made to `LoggingInterceptor`, `TestLoggingInterceptor` incorporates the changes implemented in` LoggingInterceptor`.  For the reference: `[LoggingInterceptor](driver-core/src/main/com/mongodb/internal/logging/LoggingInterceptor.java)` changes